cmd/fpga_admissionwebhook/README.md

```diff
@@ -1,97 +1,144 @@
-    certificatesigningrequest.certificates.k8s.io/intel-fpga-webhook-svc.default approved
-    secret/intel-fpga-webhook-certs created
-    Removing /tmp/tmp.XGTpddQBwP
-    Create FPGA CRDs
-    customresourcedefinition.apiextensions.k8s.io/acceleratorfunctions.fpga.intel.com created
-    customresourcedefinition.apiextensions.k8s.io/fpgaregions.fpga.intel.com created
-    fpgaregion.fpga.intel.com/arria10.dcp1.0 created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.0-compress created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.0-nlb0 created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.0-nlb3 created
-    fpgaregion.fpga.intel.com/arria10.dcp1.1 created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.1-nlb0 created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.1-nlb3 created
-    fpgaregion.fpga.intel.com/arria10.dcp1.2 created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.2-nlb0 created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.2-nlb3 created
-    fpgaregion.fpga.intel.com/d5005 created
-    acceleratorfunction.fpga.intel.com/d5005-nlb0 created
-    acceleratorfunction.fpga.intel.com/d5005-nlb3 created
-    clusterrole.rbac.authorization.k8s.io/fpga-reader created
-    clusterrolebinding.rbac.authorization.k8s.io/default-fpga-reader created
-    Create webhook deployment
-    deployment.extensions/intel-fpga-webhook-deployment created
-    Create webhook service
-    service/intel-fpga-webhook-svc created
-    Register webhook
-    mutatingwebhookconfiguration.admissionregistration.k8s.io/fpga-mutator-webhook-cfg created
+
+The script needs the CA bundle used for signing certificate requests in your cluster.
+By default, the script fetches the bundle stored in the configmap
+`extension-apiserver-authentication`. However, your cluster may use a different signing
+certificate that is passed in the option `--cluster-signing-cert-file` to `kube-controller-manager`.
+In this case, you must point the script to the actual signing certificate as follows:
+
+```bash
+$ ./scripts/webhook-deploy.sh --ca-bundle-path /var/run/kubernetes/server-ca.crt
 ```
 
+# Mappings
+
+Requested FPGA resources are translated to AF resources. For example,
+`fpga.intel.com/arria10-nlb0` is translated to `fpga.intel.com/af-d8424dc4a4a3c413f89e433683f9040b`.
+
 Mappings of resource names are configured with objects of `AcceleratorFunction` and
 `FpgaRegion` custom resource definitions found respectively in
-`./deployment/fpga_admissionwebhook/af-crd.yaml` and `./deployment/fpga_admissionwebhook/region-crd.yaml`.
+[`./deployment/fpga_admissionwebhook/af-crd.yaml`](../../deployment/fpga_admissionwebhook/af-crd.yaml)
+and [`./deployment/fpga_admissionwebhook/region-crd.yaml`](../../deployment/fpga_admissionwebhook/region-crd.yaml.)
 
 Note that the mappings are scoped to the namespaces they were created in
 and they are applicable to pods created in the corresponding namespaces.
 
```

Comment:
It might be worth pointing out to the [mapping collection](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/master/deployments/fpga_admissionwebhook/mappings-collection.yaml) that is used by webhook-deploy.sh to create mapping CRDs. There are mappings for region and AF names for different versions of Arria 10 and Stratix 10 cards. Cluster admins can add mappings for other FPGA regions and functions used in user workflows to this file and create CRDs for them using ```kubectl [--namespace <namespace>] create -f deployments/fpga_admissionwebhook/mappings-collection.yaml```